src/Storages/System/StorageSystemTables.cpp

```diff
@@ -244,10 +244,32 @@ class TablesBlockSource : public SourceWithProgress
                         if (columns_mask[src_index++])
                             res_columns[res_index++]->insert(table.second->getName());
 
-                        /// Fill the rest columns with defaults
+                        auto settings = context->getSettingsRef();
                         while (src_index < columns_mask.size())
-                            if (columns_mask[src_index++])
+                        {
+                            // total_rows
+                            if (src_index == 18 && columns_mask[src_index])
```

Comment:
Don't like hardcoded 18 and 19 because any change to columns will break this but I don't see an easy way to forward the information about the index of columns.

Adding tests should be enough then to catch it when it's affected by the above situation.

Reply:
OK,thanks,i will add test later